tests/sagemaker-tests/test_sagemaker_args.py

```diff
@@ -86,3 +90,144 @@ def test_parse_complex_string_with_quotes():
         "feat-name": "ntype0:feat0 ntype1:feat1",
     }
     assert dict(parse_unknown_gs_args(args)) == expected
+
+
+def test_get_realtime_infer_argparser():
+    """ Test the default values of realtime parser
+    """
+    test_cmd = "test_sagemaker_args.py"
+    default_args = {'--instance-type': 'ml.c6i.xlarge',
+                    '--instance-count': 1,
+                    '--model-name': 'GSF-Model4Realtime'}
+
+    # Test case 1: normal cases
+    #       1.1: for the three required arguments
+    required_args = {'--image-url': 'image_url_val',
+                     '--role': 'role_val',
+                     '--region': 'region_val'}
+    test_args = {**required_args, **default_args}
+    test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+    with mock.patch.object(sys, 'argv', test_args_str):
+        arg_parser = get_realtime_infer_parser()
+        args = arg_parser.parse_args()
+        args_w_vals = {k: v for k, v in vars(args).items() if v is not None}
+        expected_vals = {k.replace('-', '_')[2: ]: v for k, v in {**required_args, **default_args}.items()}
+        assert args_w_vals == expected_vals
+
+    #       1.2: for other arguments
+    other_args = {'--restore-model-path': 'model_path',
+                  '--model-yaml-config-file': 'yaml_config_file',
+                  '--graph-json-config-file': 'json_config_file',
+                  '--upload-tarfile-s3': 'tarfile_s3',
+                  '--infer-task-type': 'node_classification',
+                  '--model-name': 'test-model',     # here overwritten the default, and no '_'
+                  '--model-tarfile-s3': 's3://test_bucket/test.tar.gz',
+                  '--entrypoint-file-name': 'entrypoint.py'}
+    test_args = {**required_args, **default_args, **other_args}
+    test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+    with mock.patch.object(sys, 'argv', test_args_str):
+        arg_parser = get_realtime_infer_parser()
+        args = arg_parser.parse_args()
+        args_w_vals = {k: v for k, v in vars(args).items() if v is not None}
+        expected_vals = {k.replace('-', '_')[2: ]: v for k, v in test_args.items()}
+        assert args_w_vals == expected_vals
+
+    # Test case 2: abnormal cases
+    #       2.1: missing a required argument
+    for k, _ in required_args.items():
+        new_required_args = {k1: v1 for k1, v1 in required_args.items() if k1 != k}
+        test_args = {**new_required_args, **default_args}
+        test_args_str = [test_cmd] + [str(item) for pair in test_args.items() for item in pair]
+
+        with mock.patch.object(sys, 'argv', test_args_str):
+            arg_parser = get_realtime_infer_parser()
+            # parser will exit with non-zero code
+            with pytest.raises(SystemExit) as excinfo:
+                arg_parser.parse_args()
+            assert excinfo.value.code != 0
+
+    #       2.2: incorrect model_name argument
```

Comment:
The test should be moved to test_validate_realtime_infer_arguments, as the model name is checked in validate_realtime_infer_arguments.

Reply:
Not exactly, the model name is checked by the customized type checker in argeparser. The validate_realtime_infer_arguments has no code to check this.